### PR TITLE
Reflactor the vineyard scheduler code

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

As titled.

Related issue number
--------------------

Fixes some parts of https://github.com/v6d-io/v6d/issues/1234

### CODE REVIEW

File: k8s/pkg/schedulers/scheduler_on_kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	return &VineyardSchedulerOnKubernetes{
+		Client:  c,
+		pod:     pod,
+		rank:    rank,
+		replica: replica,
+	}
+}
+
+// SetupConfig setups the scheduler config
+func (vsok *VineyardSchedulerOnKubernetes) SetupConfig() error {
```
Comment (sighingnow) [inline k8s/pkg/schedulers/scheduler_on_kubernetes.go:73]: `vsok` -> `vs` as there's no `vsoc` in `k8s/pkg/schedulers/scheduler_on_kubernetes.go` so would be no confusion.

File: k8s/pkg/schedulers/scheduler_outside_cluster.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		labels:      labels,
+		config: SchedulerConfig{
+			Namespace:      namespace,
+			OwnerReference: &ownerReferences,
+		},
+	}
+}
+
+// SetupConfig setups the scheduler config
+func (vsoc *VineyardSchedulerOutsideCluster) SetupConfig() error {
```
Comment (sighingnow) [inline k8s/pkg/schedulers/scheduler_outside_cluster.go:55]: `vsoc` -> `vs` as there's no `vsok` in `k8s/pkg/schedulers/scheduler_outside_cluster.go` so would be no confusion.

**Review by sighingnow — APPROVED** [review]
Rename `VineyardSchedulerOnKubernetes` to `VineyardSchedulerInsideCluster` to make it consistent with `VineyardSchedulerOutsideCluster`?

File: docker/Dockerfile.vineyard-python-dev [hunk 3]
```diff
@@ -38,6 +38,8 @@ COPY python/client.cc /work/v6d/python/client.cc
 COPY python/core.cc /work/v6d/python/core.cc
 COPY python/error.cc /work/v6d/python/error.cc
 COPY python/pybind11_utils.cc /work/v6d/python/pybind11_utils.cc
+COPY python/pybind11_docs.cc /work/v6d/python/pybind11_docs.cc
+COPY python/pybind11_docs.h /work/v6d/python/pybind11_docs.h
 COPY python/pybind11_utils.h /work/v6d/python/pybind11_utils.h
```
Comment (sighingnow) [inline docker/Dockerfile.vineyard-python-dev:43]: Sort the filelist:

```suggestion
COPY python/pybind11_docs.cc /work/v6d/python/pybind11_docs.cc
COPY python/pybind11_docs.h /work/v6d/python/pybind11_docs.h
COPY python/pybind11_utils.cc /work/v6d/python/pybind11_utils.cc
COPY python/pybind11_utils.h /work/v6d/python/pybind11_utils.h
```

File: k8s/pkg/schedulers/scheduler_on_kubernetes.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+		vsok.config.Namespace,
+		vsok.config.OwnerReference,
+	)
+
+	target, err := bestEffort.Compute(vsok.rank)
+	if err != nil {
+		return 0, err
+	}
+
+	s, err := vsok.CheckOperationLabels()
```
Comment (sighingnow) [inline k8s/pkg/schedulers/scheduler_on_kubernetes.go:166]: Make `CheckOperationLabels` private?

File: k8s/pkg/schedulers/scheduler_outside_cluster.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	if err != nil {
+		return err
+	}
+	vsoc.config.Nodes = nodes
+
+	return nil
+}
+
+// BuildSchedulerOrder builds the scheduler order from the job to node map
+func (vsoc *VineyardSchedulerOutsideCluster) BuildSchedulerOrder(jobToNode map[string]int) string {
```
Comment (sighingnow) [inline k8s/pkg/schedulers/scheduler_outside_cluster.go:70]: Make `BuildSchedulerOrder` private?

File: k8s/pkg/schedulers/scheduler_on_kubernetes.go (hunk 1 above)
Comment (dashanji) [reply k8s/pkg/schedulers/scheduler_on_kubernetes.go:73]: Make sense to me.

**Comment by dashanji** [comment]
All comments addressed. PTAL, thanks. @sighingnow

**Pull request merged**